File: app/src/main/java/net/pside/android/rxshowdialogutil/NextActivity.java
package net.pside.android.rxshowdialogutil;

import android.content.Context;
import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;

public class NextActivity extends AppCompatActivity {

    public static Intent createIntent(Context context) {
        return new Intent(context, NextActivity.class);
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_next);
    }
}

File: app/src/main/kotlin/net/pside/android/rxshowdialogutil/NextActivity.kt
package net.pside.android.rxshowdialogutil

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity

class NextActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_next)
    }

    companion object {

        fun createIntent(context: Context): Intent {
            return Intent(context, NextActivity::class.java)
        }
    }
}
